providers/snyk/vulns_test.go

```diff
@@ -0,0 +1,208 @@
+		},
+	}
+
+	vulns, err := getVulnsForPurl("pkg:gem/tzinfo@1.2.5", newClient(&models.Credentials{}), orgID)
+
+	assert.NoError(t, err)
+	assert.Equal(t, expected, vulns)
+}
+
+func TestGetVulnsForPurlTimeout(t *testing.T) {
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	httpmock.RegisterResponder("GET", `=~\/issues\?version=`, httpmock.NewStringResponder(503, "Gateway Timeout"))
+
+	vulns, err := getVulnsForPurl("pkg:gem/tzinfo@1.2.5", newClient(&models.Credentials{}), orgID)
+
+	assert.Error(t, err)
+	assert.Equal(t, "failed request while retrieving vulnerabilities (purl: pkg:gem/tzinfo@1.2.5, status: 503)", err.Error())
+	assert.Nil(t, vulns)
+}
+
+func TestGetVulnsForPurlInvalidResponse(t *testing.T) {
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	httpmock.RegisterResponder("GET", `=~\/issues\?version=`, httpmock.NewStringResponder(200, "BOOM"))
+
+	vulns, err := getVulnsForPurl("pkg:gem/tzinfo@1.2.5", newClient(&models.Credentials{}), orgID)
+
+	assert.Error(t, err)
+	assert.Equal(t, "could not parse response (purl: pkg:gem/tzinfo@1.2.5): invalid character 'B' looking for beginning of value", err.Error())
+	assert.Nil(t, vulns)
+}
+
+func TestGetVulnsForPurlInvalidPurl(t *testing.T) {
+	httpmock.Activate()
+	defer httpmock.DeactivateAndReset()
+
+	vulns, err := getVulnsForPurl("foobar", newClient(&models.Credentials{}), orgID)
+
+	assert.Error(t, err)
+	assert.Equal(t, "invalid purl: scheme is missing", err.Error())
+	assert.Nil(t, vulns)
+}
+
+func TestSnykIssueToBomberVuln(t *testing.T) {
+	issue, err := snykIssueMock()
+	assert.NoError(t, err)
+	vuln := snykIssueToBomberVuln(issue)
+	expected := models.Vulnerability{
+		ID:          "SNYK-RUBY-TZINFO-2958048",
+		Title:       "Directory Traversal",
+		Description: "",
+		Severity:    "HIGH",
+		Cwe:         "CWE-22",
+		CvssScore:   8.1,
+		CvssVector:  "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H",
+		Reference:   "https://security.snyk.io/vuln/SNYK-RUBY-TZINFO-2958048",
+		ExternalReferences: []interface{}{
+			"https://github.com/tzinfo/tzinfo/releases/tag/v0.3.61",
+			"https://github.com/tzinfo/tzinfo/releases/tag/v1.2.10",
+			"https://github.com/tzinfo/tzinfo/commit/ca29f349856d62cb2b2edb3257d9ddd2f97b3c27",
+		},
+	}
+
+	assert.Equal(t, expected, vuln)
+}
+
+func TestSnykIssueToBomberVulnModerate(t *testing.T) {
+	issue, err := snykIssueMock()
+	assert.NoError(t, err)
+	issue.Attributes.EffectiveSeverityLevel = "medium"
+
+	vuln := snykIssueToBomberVuln(issue)
+
+	assert.Equal(t, "MODERATE", vuln.Severity)
+}
+
+func TestSnykIssueToBomberVulnMissingCwe(t *testing.T) {
+	issue, err := snykIssueMock()
+	assert.NoError(t, err)
+	issue.Attributes.Problems = []Problem{}
+
+	vuln := snykIssueToBomberVuln(issue)
+
+	assert.Equal(t, "", vuln.Cwe)
+}
+
+func TestSnykIssueToBomberVulnSnykSeverity(t *testing.T) {
+	tc := []struct {
+		Title             string
+		Severities        []Severity
+		ExpectedCvssScore float64
+	}{
+		{
+			"prefer NVD score",
```

Comment:
@garethr the code will pick NVD first, then Snyk, then others. Let me know if we should prefer Snyk sev evaluation. 🙏

Reply:
We should prefer Snyk first.

It's a good conversation to have later with the Bomber team about the nuance around multiple scores/sources/